.circleci/config.yml

```diff
@@ -928,6 +938,17 @@ jobs:
           paths:
             - development/ts-migration-dashboard/build/final
 
+  rerun-workflows-from-failed:
+    executor: node-browsers-small
+    steps:
+      - run: *shallow-git-clone-and-enable-vnc
+      - run: sudo corepack enable
+      - attach_workspace:
+          at: .
+      - run:
+          name: Rerun workflows from failed
+          command: npx tsx .circleci/scripts/rerun-ci-workflow-from-failed.ts
```

Comment:
This will tell npm to download the latest version of `tsx`, and it will install it with install scripts enabled. To ensure compatibility between versions and prevent unexpected breakages, we would want to ensure `tsx` is set to a specific known working version. We'd also want to run no install scripts by default, just those that we know to be safe (which is what `@lavamoat/allow-scripts` does).

Could we use `yarn` here, and make this depend upon `prep-deps`? Then our standard lockfile and `allow-scripts` setup would solve this problem for us.

If you think it's useful to skip the full install, there are ways we could do that too, but it would be a bit more effort.

Reply:
thank you for your elaborated comment, Mark :pray: mmm I see.. I think I get what you mean, and that I've addressed your comment, could you let me know if the new changes make sense to you?

I've also tested them (disabling temporarily the filters), and it works fine:

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/111692/workflows/d72ea93c-fc27-4c89-9da0-0b40bce4c4eb/jobs/4181360

![Screenshot from 2024-11-20 19-46-23](https://github.com/user-attachments/assets/24fbe87d-2cfe-4969-b2e0-4adf9214b580)